### PR TITLE
Add networkd notify

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

I want to have this... so I started to implement a package for it.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is one of the first services I write, so please tell me what to improve etc. It is not tested yet, though (as I do not know how to).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Waiting for https://github.com/wavexx/networkd-notify/issues/1

File: pkgs/applications/misc/networdk-notify/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "006d8qmmk0jg0a3fbl2q934wmmhx9n7c156i34i9zgzapdhhacq6";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [ dbus-python ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ls
```
Comment (Mic92) [inline pkgs/applications/misc/networdk-notify/default.nix]: leftover?

File: pkgs/applications/misc/networdk-notify/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, pythonPackages, fetchFromGitHub }:
+
+pythonPackages.buildPythonPackage rec {
+  name = "networkd-notify-${version}";
```
Comment (Mic92) [inline pkgs/applications/misc/networdk-notify/default.nix]: Please use the following notation for unstable packages:

```
name = "networkd-notify-unstable-2016-05-17";
```

File: pkgs/applications/misc/networdk-notify/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  propagatedBuildInputs = with pythonPackages; [ dbus-python ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ls
+    cp networkd-notify $out/bin/
+    chmod +x $out/bin/networkd-notify
```
Comment (Mic92) [inline pkgs/applications/misc/networdk-notify/default.nix]:
```
install -m755 networkd-notify $out/bin/networkd-notify
```

File: nixos/modules/services/desktops/networkd-notify.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  ###### implementation
+
+  config = mkIf config.services.networkd-notify.enable {
+
+    environment.systemPackages = [ pkgs.networkd-notify ];
+
+    systemd.services.networkd-notify = {
+      description = "networkd desktop notification";
+      wantedBy    = [ "multi-user.target" ];
```
Comment (Mic92) [inline nixos/modules/services/desktops/networkd-notify.nix]: Does a system service running as root find a user dbus session?

Comment (matthiasbeyer) [inline nixos/modules/services/desktops/networkd-notify.nix]: Yes, but I don't know how to do it :-)

**Comment by matthiasbeyer** [comment]
I also need some help with the package itself. I was told that "gi" (the dependency) is the "python Gobject-Introspection package", though I do not find it in nixpkgs (python-gobject2/3 does not work). Ideas?

File: pkgs/applications/misc/networdk-notify/default.nix [hunk 5]
```diff
@@ -16,6 +16,7 @@ pythonPackages.buildPythonPackage rec {
   phases = [ "unpackPhase" "installPhase" ];
 
   installPhase = ''
+    mkdir -p $out/bin/
     install -m755 networkd-notify $out/bin/networkd-notify
```
Comment (Mic92) [inline pkgs/applications/misc/networdk-notify/default.nix]: you can also use "-D" for install instead of the "mkdir".

**Comment by matthiasbeyer** [comment]
Done. I still don't know what the "python-gobject" thing should be ... so there's still a dependency missing here.

**Comment by 7c6f434c** [comment]
Hm, maybe factor out just the package so that it can be merged? Or is the service also working by now?

**Comment by matthiasbeyer** [comment]
Stalled.

**Pull request closed without merging**